list-agreements.lua

```diff
@@ -1,120 +1,218 @@
+Lists outstanding Guildhall and Temple agreements in fortress mode.
 
-]====]
+Additionally:
+
+
+* Translated names of the associated Orders and Guilds
+* Worshiped Deitys and Professions respectively
+* Petition age and status satisfied, denied or expired, or blank for outstanding
+
+
+
+Arguments
+---------------
+
+    all     list all agreements; past and present
 
+    help    script help
+
+]====]
 local playerfortid = df.global.ui.site_id -- Player fortress id
 local templeagreements = {} -- Table of agreements for temples in player fort
 local guildhallagreements = {} -- Table of agreements for guildhalls in player fort
 
-function get_location_tier(agr)
-    loctier = agr.details[0].data.Location.tier
-    return loctier
+function get_location_name(loctier,loctype)
+    local locstr = "Unknown Location"
+    if loctype == df.abstract_building_type.TEMPLE and loctier == 1 then
+        locstr = "Temple"
+    elseif loctype == df.abstract_building_type.TEMPLE and loctier == 2 then
+        locstr = "Temple Complex"
+    elseif loctype == df.abstract_building_type.GUILDHALL and loctier == 1 then
+        locstr = "Guildhall"
+    elseif loctype == df.abstract_building_type.GUILDHALL and loctier == 2 then
+        locstr = "Grand Guildhall"
+    end
+    return locstr
 end
 
 function get_location_type(agr)
     loctype = agr.details[0].data.Location.type
     return loctype
 end
 
-function get_location_name(agr)
-    if get_location_type(agr) == df.abstract_building_type.TEMPLE and get_location_tier(agr) == 1 then
-        return "temple"
-    elseif get_location_type(agr) == df.abstract_building_type.TEMPLE and get_location_tier(agr) == 2 then
-        return "temple complex"
-    elseif get_location_type(agr) == df.abstract_building_type.GUILDHALL and get_location_tier(agr) == 1 then
-        return "guildhall"
-    elseif get_location_type(agr) == df.abstract_building_type.GUILDHALL and get_location_tier(agr) == 2 then
-        return "grand guildhall"
-    end
+function get_petition_date(agr)
+    local agr_year = agr.details[0].year
+    local agr_year_tick = agr.details[0].year_tick
+    local julian_day = math.floor(agr_year_tick / 1200) + 1
+    local agr_month = math.floor(julian_day / 28) + 1
+    local agr_day = julian_day % 28
+    return string.format("%03d-%02d-%02d",agr_year, agr_month, agr_day)
 end
 
-function get_petition_date(agr)
-    agr_year = agr.details[0].year
-    agr_year_tick = agr.details[0].year_tick
-    julian_day = math.floor(agr_year_tick / 1200) + 1
-    agr_month = math.floor(julian_day / 28) + 1
-    agr_day = julian_day % 28
-    return agr_year.."-"..agr_month.."-"..agr_day
+function get_petition_age(agr)
+    local agr_year_tick = agr.details[0].year_tick
+    local agr_year = agr.details[0].year
+    local cur_year_tick = df.global.cur_year_tick
+    local cur_year = df.global.cur_year
+    local del_year, del_year_tick, del_month, del_day
```

Comment:
since `del_month` and `del_day` aren't used until after the if blocks, I'd suggest just declaring them `local` below and not here.